content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -56,8 +56,8 @@ Kubernetes implements horizontal pod autoscaling as a control loop that runs int
 (and the default interval is 15 seconds).
 
 Once during each period, the controller manager queries the resource utilization against the
-metrics specified in each HorizontalPodAutoscaler definition.  The controller manager
-obtains the metrics from either the resource metrics API (for per-pod resource metrics),
+metrics specified in each HorizontalPodAutoscaler definition.  The controller manager selects
+the pods based on the `scaleTargetRef` resource's labels and obtains the metrics from either the resource metrics API (for per-pod resource metrics),
```

Comment:
I'm not 100% sure this is right.
I think what happens is that the HPA controller selects Pods based on the `.spec.selector` of the resource that `scaleTargetRef` points at.

It's a small detail because `.spec.selector` is always a _label selector_ and not a _field selector_ - that's how today's API is defined. However, I think people might think that the labels for (eg) a StatefulSet matter when matching Pods for scaling, whereas in fact it's the labels in that StatefulSet's pod template and the selector for that StatefulSet that matter more.

@guymenahem & @jimangel is that worth clarifying?

Reply:
You are right, It's part of the `.spec.selector`, I thought that the link between these is clear enough but the better way is to not leave any place for ambiguity

I will fix it.

Reply:
@sftim - I added the full process of selection.
I think that it will be easier for the readers to understand what to look for in their configuration or how to configure it properly.